lib/fizzy/parser_expr.cpp

```diff
@@ -10,23 +10,30 @@ inline void push(bytes& b, T value)
     b.resize(b.size() + sizeof(value));
     __builtin_memcpy(&b[b.size() - sizeof(value)], &value, sizeof(value));
 }
+
+/// The immediates offset for block instructions.
+using label_position = size_t;
+
+/// Sentinel constant to loop labels. They are not interested for parser.
+constexpr auto loop_label = label_position(-1);
+
 }  // namespace
 
 parser_result<Code> parse_expr(const uint8_t* pos)
 {
     Code code;
-    Instr instr;
-    do
+    std::vector<label_position> label_positions;
```

Comment:
I would prefer a comment here, too, saying that these are positions in immediates